uui-core/src/helpers/range.ts

```diff
@@ -0,0 +1,26 @@
+function* rangeFromTo(start: number, end: number, step: number = 1) {
+    for (let i = start; i < end; i = i + step) {
+        yield i;
+    }
+}
+
+function* reverseRangeFromTo(start: number, end: number, step: number = 1) {
+    for (let i = start; i > end; i = i - step) {
+        yield i;
+    }
+}
+
+function* rangeGen(start: number, end?: number, step: number = 1) {
+    if (end === undefined) {
+        yield* rangeFromTo(0, start, step);
+    }
+
+    if (start < end) {
+        yield* rangeFromTo(start, end, step);
+    } else {
+        yield* reverseRangeFromTo(start, end, step);
+    }
+}
+
+export const range = (start: number, end?: number, step: number = 1) =>
+    Array.from(rangeGen(start, end, step));
```

Comment:
Can't we return iterators from 'range' function, w/o converting them to array?

JS-Iterator-based API for 'range' can be a good idea, as in this case we don't create an array just to iterate over it 
indexes and throw it away.

If we return array anyway, it will be simpler and faster to go old-school like:

const result = new Array(end - start); // to allocate space for items
for(var n = start; n < end; n++) arr[n] = n;

Reply:
The previous version returned a generator only, but all over the project `.map/.forEach` Array methods are used.
That's why I've returned an array from generator.
But, probably, we can make it simpler by generating array.